numba/tests/test_ufuncs.py

```diff
@@ -751,38 +753,42 @@ def test_broadcasting(self):
             self.assertPreciseEqual(result, expected)
 
     def test_implicit_output_npm(self):
+        # This is a test for Issue #1078
+        # (https://github.com/numba/numba/issues/1078) It ensures that the
+        # implicit output is an array as opposed to a scalar by attempting to
+        # return the result of a ufunc with nrt disabled, which should cause an
+        # error (because we cannot return an array without nrt)
+        def myadd(a0, a1):
+            return np.add(a0, a1)
+        arr_ty = types.Array(types.uint64, 1, 'C')
+
         with self.assertRaises(NumbaTypeError):
-            def myadd(a0, a1):
-                return np.add(a0, a1)
-            arr_ty = types.Array(types.uint64, 1, 'C')
-            cr = compile_isolated(myadd, (arr_ty, arr_ty),
-                                  flags=no_pyobj_flags)
+            njit((arr_ty, arr_ty), _nrt=False)(myadd)
```

Comment:
Should this use `self._compile` ?

Reply:
I'd say it shouldn't - the result won't be cached because it's expected to fail, and it's more transparent what's going on in the test when the `njit` function is used directly.

Reply:
ah I see, thanks, makes sense to leave as-is.

Reply:
Well, as we concluded OOB, this is just daft, so https://github.com/numba/numba/pull/8625/commits/9ecfe70d3408836dc72b30a43fbded52b326c2ea changes the test to not need to disable NRT. I've still left the use of `njit` here so it's clear what's going on.